run_telescope.sh

```diff
@@ -0,0 +1,264 @@
+        warning "Configuration file already exists. Skipping."
+    else
+        cat <<EOF > "${TELESCOPE_CONFIG_FILE}"
+server:
+  log_level: info
+
+metrics:
+  wal_directory: /tmp/wal
+  global:
+    scrape_interval: 15s
+    external_labels:
+      project_id: ${PROJECT_ID}
+      project_name: ${PROJECT_NAME}
+    remote_write:
+      - url:
+        basic_auth:
+          username: 
+          password: 
+
+  configs:
+EOF
+        case "${NETWORK}" in
+            ethereum)
+                cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+    - name: geth
+      scrape_configs:
+        - job_name: geth
+          static_configs:
+            - targets: ["localhost:9100"]
+EOF
+                ;;
+            polkadot)
+                cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+    - name: parachain
+      scrape_configs:
+        - job_name: parachain
+          static_configs:
+            - targets: ["localhost:9615"]
+    - name: relaychain
+      scrape_configs:
+        - job_name: relaychain
+          static_configs:
+            - targets: ["localhost:9616"]
+EOF
+                ;;
+            arbitrum)
+                cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+    - name: arbitrum
+      scrape_configs:
+        - job_name: arbitrum
+          static_configs:
+            - targets: ["localhost:9100"]
+EOF
+                ;;
+            base)
+                cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+    - name: base
+        scrape_configs:
+            - job_name: base
+                static_configs:
+                - targets: ["localhost:9100"]
+EOF
+                ;;
+            optimism)
+                cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+    - name: optimism
+        scrape_configs:
+            - job_name: optimism
+                static_configs:
+                - targets: ["localhost:9100"]
+EOF
+                ;;
+        esac
+        cat <<EOF >> "${TELESCOPE_CONFIG_FILE}"
+integrations:
+  agent:
+    enabled: false
+  node_exporter:
+    enabled: true
+EOF
+        success "Configuration file created successfully."
+    fi
+}
+
+function get_docker_image () {
+    echo -n "Pulling docker image...  "
+    TELESCOPE_IMAGE=grafana/agent:v0.37.2
+    docker pull --quiet "${TELESCOPE_IMAGE}"
+    if [[ 0 -ne $? ]]
+    then
+        error "Failed to pull the telescope docker image."
+    fi
+
+    info "OK"
+}
+
+function create_docker_volume () {
+    echo -n "Creating docker volume...  "
+    TELESCOPE_VOLUME=telescope
+    docker volume create "${TELESCOPE_VOLUME}" &> /dev/null
+    if [[ 0 -ne $? ]]
+    then
+        warning "Docker volume already exists. Skipping."
+    fi
+
+    info "OK"
+}
+
+function run_telescope () {
+    echo -n "Setting up Telescope...  "
+    TELESCOPE_CONTAINER=telescope
+    docker run -d --rm \
+        --name "host" --network="host" --pid="host" --cap-add SYS_TIME \
+        --mount "type=volume,source=${TELESCOPE_VOLUME},destination=/var/lib/grafana-agent" \
```

Comment:
given teh variable `TELESCOPE_VOLUME ` isn't declared in this function neither is it globally.. i don't think it would work here unless we decalre it, most likley as a global variable like PROJECT_ID above